pkg/crypto/keys/publickey.go

```diff
@@ -98,11 +98,20 @@ type PublicKey ecdsa.PublicKey
 
 // Equal returns true in case public keys are equal.
 func (p *PublicKey) Equal(key *PublicKey) bool {
-	return p.X.Cmp(key.X) == 0 && p.Y.Cmp(key.Y) == 0
+	return p.Cmp(key) == 0
 }
 
 // Cmp compares two keys.
 func (p *PublicKey) Cmp(key *PublicKey) int {
+	if p.X == nil || p.Y == nil {
```

Comment:
The case is that we can't use `||` here. We can only operate with `IsInfinity` definition when both coordinates are not set, otherwise the key is invalid. So we just need to check if one of `p` or `key` is infinity and return the result based on this. See how it's done in the reference implementation:
https://github.com/neo-project/neo/blob/9b9be47357e9065de524005755212ed54c3f6a11/src/Neo/Cryptography/ECC/ECPoint.cs#L66-L67